src/exampleTypes/TagsText.js

```diff
@@ -23,41 +23,41 @@ let Text = ({
   tree,
   node,
   placeholder,
-  theme: { Select, TagsInput, Popover },
-}) => {
-  let open = React.useState(false)
-  let [selectedTag, setSelectedTag] = React.useState(null)
-  return (
-    <div className="contexture-text">
-      <Select
-        value={node.operator}
-        onChange={e => tree.mutate(node.path, { operator: e.target.value })}
-        options={operatorOptions}
-      />
-      <TagsInput
-        splitCommas
-        tags={node.values}
-        onTagClick={tag => {
-          F.on(open)()
-          setSelectedTag(tag)
-        }}
-        addTag={tag => {
-          tree.mutate(node.path, { values: [...node.values, tag] })
-        }}
-        removeTag={tag => {
-          tree.mutate(node.path, {
-            values: _.without([tag], node.values),
-          })
-        }}
-        tagStyle={bgJoin(tagToGroupJoin(node.join))}
-        submit={tree.triggerUpdate}
-        placeholder={placeholder}
-      />
-      <Popover open={open}>
-        <TagsJoinPicker tag={selectedTag} node={node} tree={tree} />
-      </Popover>
-    </div>
-  )
-}
+  theme: { Select, Tag, TagsInput, Popover },
+}) => (
+  <div className="contexture-text">
+    <Select
+      value={node.operator}
+      onChange={e => tree.mutate(node.path, { operator: e.target.value })}
+      options={operatorOptions}
+    />
+
+    <TagsInput
+      splitCommas
+      tags={node.values}
+      addTag={tag => {
+        tree.mutate(node.path, { values: [...node.values, tag] })
+      }}
+      removeTag={tag => {
+        tree.mutate(node.path, {
+          values: _.without([tag], node.values),
+        })
+      }}
+      Tag={props => (
+        <Popover
+          trigger={<Tag {..._.omit('onClick', props)} />}
```

Comment:
Won't the click bubble up and get picked up by the popover anyway?

Reply:
Without filtering our `onClick` popover would open only on the second click. I am working to change this implementation